File: apps/backend/app/Domain/Reservation/Entities/Reservation.php
<?php

namespace App\Domain\Reservation\Entities;

use App\Domain\Reservation\Enums\ReservationStatus;

final readonly class Reservation
{
    public function __construct(
        public string $id,
        public string $tenantId,
        public string $clientId,
        public ?string $clientResourceId,
        public string $serviceId,
        public ?string $assignedTo,
        public \DateTimeImmutable $scheduledAt,
        public \DateTimeImmutable $estimatedEnd,
        public ReservationStatus $status,
        public ?string $notes,
        public ?\DateTimeImmutable $cancelledAt,
        public ?string $cancelReason,
        public string $createdBy,
    ) {}
}

File: apps/mobile/lib/features/reservations/domain/entities/reservation.dart
import '../enums/reservation_status.dart';

class Reservation {
  final String id;
  final String clientResourceId;
  final String serviceId;
  final String? assignedTo;
  final DateTime scheduledAt;
  final DateTime estimatedEnd;
  final ReservationStatus status;
  final String? notes;
  final String? clientResourceLabel;
  final String? serviceName;
  final String? servicePrice;
  final String? clientName;

  const Reservation({
    required this.id,
    required this.clientResourceId,
    required this.serviceId,
    this.assignedTo,
    required this.scheduledAt,
    required this.estimatedEnd,
    required this.status,
    this.notes,
    this.clientResourceLabel,
    this.serviceName,
    this.servicePrice,
    this.clientName,
  });
}
